### PR TITLE
Refactor repo

### PR DESCRIPTION
**Pull request by rshewitt**
# Pull Request

Related to [#4502](https://github.com/GSA/data.gov/issues/4502)

## About

Removed the flask controller, airflow implementation, sqlalchemy, postgres, s3, and refactored the remaining functionality. We can `git revert` anything we feel needs to be included in the future. This work is intended to remove anything that we don't immediately need within the context of using this repo as a python module in a workflow orchestration system like airflow.  

## PR TASKS

- [x] The actual code changes.
- [x] Tests written and passed.
- [x] Any changes to docs?
- [ ] Bumped version number in [pyproject.toml](https://github.com/GSA/datagov-harvesting-logic/blob/8078c5b9f015ca7fb8a142e4e4b2e22ad2fd49b1/pyproject.toml#L3) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: harvester/utils/json.py [hunk 1]
```diff
@@ -1,38 +1,11 @@
...
-
-    if resp.status_code != 200:
-        raise Exception("non-200 status code")
-
-    try:
-        data = resp.json()
-    except json.JSONDecodeError as e:
-        raise Exception(e)
-
-    return data
```
Comment (jbrown-xentity) [inline harvester/utils/json.py:38]: I don't think we want to remove this code; this will be needed in the final version of this library.

**Review by jbrown-xentity — COMMENTED** [review]
Close, I would keep one piece of this... Otherwise it's ready!

File: harvester/utils/json.py (hunk 1 above)
Comment (rshewitt) [reply harvester/utils/json.py:38]: okay I'll add that back in.

Comment (rshewitt) [reply harvester/utils/json.py:38]: added in bea96d7aef757d3168899a1e4e23d882872e7f7a

**Review by robert-bryson — APPROVED** [review]

**Pull request merged**